testing/python/metafunc.py

```diff
@@ -1503,6 +1548,31 @@ def test_it(x): pass
         result = pytester.runpytest()
         assert result.ret == 0
 
+    def test_parametrize_module_level_test_with_class_scope(
+        self, pytester: Pytester
+    ) -> None:
+        module = pytester.makepyfile(
+            """
+            import pytest
+
+            @pytest.mark.parametrize("x", [0, 1], scope="class")
+            def test_1(x):
+                pass
+
+            @pytest.mark.parametrize("x", [1, 2], scope="module")
+            def test_2(x):
+                pass
+        """
+        )
+        test_1_0, _, test_2_0, _ = pytester.genitems((pytester.getmodulecol(module),))
+        test_1_fixture_x = cast(Function, test_1_0)._fixtureinfo.name2fixturedefs["x"][
+            -1
+        ]
+        test_2_fixture_x = cast(Function, test_2_0)._fixtureinfo.name2fixturedefs["x"][
+            -1
+        ]
```

Comment:
Instead of `cast`, better to assert, which also makes sure what we expect is actually the case.

```suggestion
        assert isinstance(test_1_0, Function)
        assert isinstance(test_2_0, Function)
        test_1_fixture_x = test_1_0._fixtureinfo.name2fixturedefs["x"][-1]
        test_2_fixture_x = test_2_0._fixtureinfo.name2fixturedefs["x"][-1]
```